string name
string id
string message